airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -0,0 +1,1152 @@
+    let domain = getOrCreateAirDomain(new Domain(domainId, chainId, block, tokenAddress));
+    if (fromOldRegistry && domain.isMigrated == true) {
+      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
+      return;
+    }
+    let previousOwnerId = domain.owner;
+    let parent = getOrCreateAirDomain(new Domain(
+      node.toHexString(),
+      chainId,
+      block,
+      tokenAddress,
+    ));
+    if (domain.parent == null && parent != null) {
+      parent.subdomainCount = parent.subdomainCount.plus(BIGINT_ONE);
+    }
+    if (domain.name == null) {
+      let labelName = ens.nameByHash(label.toHexString());
+      if (labelName != null) {
+        domain.labelName = labelName;
+      }
+      if (labelName === null) {
+        labelName = '[' + label.toHexString().slice(2) + ']';
+      }
+      if (node.toHexString() == ROOT_NODE) {
+        domain.name = labelName;
+      } else {
+        let name = parent.name;
+        if (labelName && name) {
+          domain.name = labelName + '.' + name;
+        }
+      }
+    }
+
+    let tokenId = BigInt.fromUnsignedBytes(label).toString();
+    domain.owner = getOrCreateAirAccount(chainId, newOwner).id;
+    domain.parent = parent.id;
+    domain.labelHash = label;
+    domain.isMigrated = isMigrated;
+    domain.tokenId = tokenId;
+    domain.lastBlock = block.id;
+    recurseSubdomainCountDecrement(domain, chainId, block, tokenAddress);
+    domain.save();
+
+    // creating reverse registrar to get domainId when setting primary domain
+    if (domain.name) {
+      createReverseRegistrar(domain.name!, domain.id, block);
+    }
+
+    getOrCreateAirDomainOwnerChangedTransaction(
+      block,
+      logIndex,
+      chainId,
+      previousOwnerId,
+      newOwner,
+      transactionHash,
+      tokenId,
+      domain,
+    );
+  }
+
+  /**
+   * @dev This function tracks a domain transfer transaction
+   * @param node specifies the node of the domain
+   * @param chainId chain id
+   * @param newOwnerAddress specifies the new owner of the domain
+   * @param blockHeight block number in the chain
+   * @param blockHash block hash
+   * @param blockTimestamp block timestamp
+   * @param logIndex txn log index
+   * @param transactionHash transaction hash
+   * @param tokenAddress contract address of nft token
+   * @param fromOldRegistry specifies if the event is from the old registry
+   */
+  export function trackDomainTransferTransaction(
+    node: string,
+    chainId: string,
+    newOwnerAddress: string,
+    blockHeight: BigInt,
+    blockHash: string,
+    blockTimestamp: BigInt,
+    logIndex: BigInt,
+    transactionHash: string,
+    tokenAddress: string,
+    fromOldRegistry: boolean,
+  ): void {
+    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
+    let domain = getOrCreateAirDomain(new Domain(node, chainId, block, tokenAddress));
```

Comment:
the domain entity has some required fields which have to be set with a default value when creating.
and the data to update this entity is scattered in a lot of different events. that's why the domain entity updation code is being written after a default domain entity has been created, otherwise the subgraph sync will keep on failing.

Reply:
updates to the domain entity are done inside the track funcs

Reply:
@rahul7668gupta  To fix this.

Reply:
updated owner for a domain in #39